src/main/java/com/mackervoy/calum/mud/content/ContentController.java

```diff
@@ -25,30 +31,50 @@
  */
 @Path("/content/")
 public class ContentController {
-  @GET
-  @Produces("text/turtle")
-  public String getContent(@QueryParam("uri") String uri) {
-		final String safeUri = uri == null ? "" : uri;
-		final Model m = ModelFactory.createDefaultModel();
-    m.read(safeUri);
-
-		return getDescriber(safeUri, m)
-			.flatMap(describer -> describer.describe(m, safeUri))
-      .map(contentModel -> modelToTurtle(contentModel))
-			.orElse(null);
-  }
+	//NOTE: the ContentContoller POST must receives objects with RDF type set, or it will ignore them
+	@POST
+	public Response post(String requestBody) {
+		final Model request = ModelFactory.createDefaultModel();
+		request.read(new StringReader(requestBody), "", "TURTLE");
+	
+		//build the result model by iterating over each object in the scene and annotating a description
+		//TODO: for now we are just describing everything, later we will want to be able to optimise what is described
+		Model result = ModelFactory.createDefaultModel();
+		ResIterator resources = request.listResourcesWithProperty(RDF.type);
+		while(resources.hasNext()) {
+			Resource res = resources.next();
+			Model m = ModelFactory.createDefaultModel();
+			m.read(res.getURI());
+			final Resource r = m.getResource(res.getURI());
+			
+			System.out.println(r);
+			Model content = this.getDescriber(r)
+					.flatMap(describer -> describer.describe(r))
+					.orElse(null);
+			
+			if(content != null) result.add(content);
```

Comment:
Instead of the check at the end if it's null, you can use `.isPresent(content -> result.add(content))` - that'd let you skip the `orElse` bit as well :smile:

Reply:
Ah yeah this is cleaner, thanks